### ISSUE TITLE
Running `outlier_detection` with `in_memory=False` reads all models into memory

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#8478]
_Issue [JP-3619](https://jira.stsci.edu/browse/JP-3619) was created on JIRA by [Brett Graham](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bgraham):_

Running outlier detection with the recent main commit: 13e092743d6ccf9ce5352ea4d8899b726532aa4a

With an association containing 33 members (with no custom grouping or tweakreg_catalog entries for any member).

And with "in_memory=False" results in the following memory usage:

The attached graph shows that even with "in_memory=False" all models are read into memory (the largest plateau in the graph). The main cause is that the implementation of ModelContainer.models_grouped  (which is used in resample) opens all models and keeps them in memory.

**Comment by jdavies-st** [comment]
I will add here that when `in_memory=True` for `outlier_detection`, it still writes out the blot files and then reads them back in.

```
2024-05-15 15:48:21,169 - stpipe.Image3Pipeline.outlier_detection - INFO - Blotting (2048, 2048) <-- (6321, 9320)
2024-05-15 15:48:21,455 - stpipe.Image3Pipeline.outlier_detection - INFO - Saved model in jw02321001009_02101_00006_nrcblong_c1001_blot.fits
```

So it is writing to disk when it should be doing everything in memory.  This is with release 1.14.0.

**Comment by stscijgbot-jp** [comment]
_Comment by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter) on [JIRA](https://jira.stsci.edu/browse/JP-3619?focusedCommentId=711747#comment-711747):_

This should be fixed as part of implementing ModelLibrary (see [JP-3690](https://jira.stsci.edu/browse/JP-3690)).  I will try to remember to post memory profiling for outlier_detection here as well as on that ticket as I continue to develop and test the changes.

**Comment by stscijgbot-jp** [comment]
_Comment by [Melanie Clarke](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mclarke) on [JIRA](https://jira.stsci.edu/browse/JP-3619?focusedCommentId=719558#comment-719558):_

Fixed by #8683
